src/casp/services/api/data_manager/cellarium_general.py

```diff
@@ -80,31 +79,43 @@ def get_model_by_name(model_name: str) -> models.CASModel:
 
         :return: CAS ML model object from the database
         """
-        model = models.CASModel.query.filter_by(model_name=model_name).first()
+        with self.system_data_db_session_maker() as session:
+            model = session.query(models.CASModel).filter_by(model_name=model_name).first()
 
-        if model is None:
-            raise exceptions.NotFound(f"Model {model_name} not found in the database")
+            if model is None:
+                raise exceptions.NotFound(f"Model {model_name} not found in the database")
 
-        return model
+            return model
 
-    @classmethod
-    def get_index_for_model(cls, model_name: str) -> models.CASMatchingEngineIndex:
+    def get_index_for_model(self, model_name: str) -> models.CASMatchingEngineIndex:
         """
         Retrieve CAS model index by its system name
 
         :param model_name: ML Model name
 
         :return: CAS ML model index object from the database
         """
-        return cls.get_model_by_name(model_name=model_name).cas_matching_engine
 
-    def increment_user_cells_processed(self, user: models.User, number_of_cells: int) -> None:
+        with self.system_data_db_session_maker() as session:
+            model = session.query(models.CASModel).filter_by(model_name=model_name).first()
+
+            if model is None:
+                raise exceptions.NotFound(f"Model {model_name} not found in the database")
+
+            return model.cas_matching_engine
+
+    def log_user_activity(self, user_id: int, model_name: str, method: str, cell_count: int) -> None:
         """
-        Increment `requests_processed` and `cells_processed user stats`.
+        Log the information to the DB about a request from a user
 
-        :param user: A SQLAlchemy user model
-        :param number_of_cells: number of cells to increment by
+        :param user_id: The id for a user
+        :param model_name: Model name that user is using
+        :param method: Method that user is using
+        :param cell_count: Number of cells processed in this request
         """
-        user.requests_processed += 1
-        user.cells_processed += number_of_cells
-        self.system_data_db_session.commit()
+        user_activity = models.UserActivity(
+            user_id=user_id, model_name=model_name, method=method, cell_count=cell_count
+        )
+        with self.system_data_db_session_maker() as session:
```

Comment:
Just going back to `with self.system_data_db_session_maker.begin()` or `with self.system_data_db_session_maker().begin()` conversation, do you think we can use it here to minimize number of nested calls?

Reply:
Just for the context:
https://docs.sqlalchemy.org/en/14/orm/session_api.html#sqlalchemy.orm.sessionmaker.begin

You can use `begin` directly from sessionmaker.

Reply:
Oh cool, that's much cleaner.  I like that better.  I've made that change